docs/infra/treatment-service.md

```diff
@@ -0,0 +1,132 @@
+# Treatment Service Deployment
+
+## Overview
+XP's Treatment Service is currently offered in two different versions:
+
+- a central Treatment Service, *deployed as a single deployment, to serve treatments to multiple 
+  clients/users with multiple projects*
+- a Treatment Service plugin, *deployed as a Turing Router experiment engine plugin, to serve treatments to a single 
+  client/user with a single project*
```

Comment:
nit: `within a single CaraML project`

Reply:
Shall we link the Turing repo to the word "Turing Router" in the description?

Reply:
Added `CaraML` and the link to the Turing repo!